integrationTests/DonationRepositoryTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace MatchBot\IntegrationTests;
+
+use Doctrine\ORM\EntityManagerInterface;
+use MatchBot\Application\HttpModels\DonationCreate;
+use MatchBot\Domain\Campaign;
+use MatchBot\Domain\Charity;
+use MatchBot\Domain\Donation;
+use MatchBot\Domain\DonationRepository;
+use MatchBot\Domain\DonationStatus;
+use MatchBot\Domain\PaymentMethodType;
+
+class DonationRepositoryTest extends IntegrationTest
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->clearPreviousCampaignsCharitiesAndRelated(); // Avoid e.g. HMRC ref dupes
```

Comment:
It feels a bit odd to do this specifically for _this_ repo test. But I also feel like because the fixture charity statuses here are very specific to what this class/method test is targeting, it's easier to understand if the test data is set up in this class too.

Very open to any ideas on how to get the best of both approaches.

Reply:
What if we randomise the ref and anything else that could conflict? Maybe then we wouldn't need to delete anything from the DB.

I think we don't have a separate database for manual testing and automated testing in dev environments, so clearing this stuff might cause problems for manual testing.

Reply:
Yeah, I think I'd misremembered what happened after I pointed out the risks with the database not being clearly separated out. I think I thought we'd made a new named schema, or at least restricted the app_env further to `test`, but if that's not the case I agree this could lead to unhelpful dev mishaps.

I'm really not a fan of the test data expanding indefinitely, especially as I have dozens of projects and run out of Docker space fairly regularly. I think if we could isolate the tests better and reset their DB regularly it would make it less likely we were relying on local side effects and would mean better dev quality of life around these tests.

Reply:
Right - I think a bit of expansion is OK, it's sometimes handy to look in the DB after running some tests to see what was left, and the data added here is going not going to take much space at all.

Or we maybe add a WHERE to the delete so we only delete the records created by this test instead of everything?

Not an absolute blocker if you want to merge with deleting everything, but I might change it later if I find it too annoying having the charities cleared on running integration tests.

Reply:
Let's leave as is for now, consider making a separate schema later.